### ISSUE TITLE
BUG: Mixpanel populating wrong location in the user profile

### ISSUE THREAD

**Issue by akbansa** [rudderlabs/rudder-transformer#3036]
**Describe the bug**
When we send **identify()** calls to Rudderstack, the transformed data removes the IP (in case of `$merge` calls) that leads to the wrong location in the Mixpanel. 

**To Reproduce**
- Add a mixpanel integration with NodeJs
- Send an `identify()` call with user IP in the context
- When the anonymous ID is merged with the user (during `$merge`), it just sends the anonymous ID and user ID in the payload.
- This leads to mixpanel picking up the Rudderstack IP and updating the user profile with the Ashburn, US location (i.e., RS server location)

**Expected behavior**
It should send the IP in the `identify()` calls to fix the wrong location update in Mixpanel's user profile.

**Screenshots**
<img width="915" alt="Screenshot at Jan 29 12-55-23" src="https://github.com/rudderlabs/rudder-transformer/assets/13042781/d804df0f-efde-4cbe-b3f9-b447b61c5cc2">
<img width="411" alt="Screenshot at Jan 29 12-55-50" src="https://github.com/rudderlabs/rudder-transformer/assets/13042781/1c1c3c45-6631-402f-9e34-d8245333ad75">

**Additional context**
[I could find the problem here](https://github.com/rudderlabs/rudder-transformer/blob/db2c1703ec0daa34d44604acc06f592c3cf44521/src/v0/destinations/mp/transform.js#L281)

**Comment by ItsSudip** [comment]
Hey @akbansa,

We tried to reproduce this but unfortunately were not able to reproduce it. We have done the following set of calls
1. Identify call with anonymousId only along with IP in the context.
2. Same identify call but added userId.

Can you share in which sequence you are sending the calls?

We are using [this](https://developer.mixpanel.com/reference/identity-merge) API so we can't send IP addresses there as there are only scopes for disctinct_ids. 

And if you want to get rid of merge call you can use the [simplified ID merge](https://www.rudderstack.com/docs/destinations/streaming-destinations/mixpanel/#:~:text=If%20you%20select%20the%20Simplified%20ID%20Merge%20option%2C%20make%20sure%20to%20turn%20on%20the%20Simplified%20ID%20Merge%20API%20for%20your%20project%20before%20sending%20the%20events.).
Thanks.

**Comment by akbansa** [comment]
Hey @ItsSudip 

Thank you for checking it quickly and responding. Let me try againa and come up with the steps.

**Comment by gitcommitshow** [comment]
@akbansa do you still face this issue?

**Comment by akbansa** [comment]
@gitcommitshow I think, we can close it for now. We are still facing this issue but I'm sure it's not by RudderStack.

**Comment by sanpj2292** [comment]
Closing the issue for now.
